dashboard/src/components/TreeListingPage/TreeListingPage.tsx

```diff
@@ -1,81 +1,107 @@
-import { useSearch } from '@tanstack/react-router';
-
 import { useMemo, useState } from 'react';
 
 import { FormattedMessage } from 'react-intl';
 
-import { zOrigin, TreeTableBody } from '@/types/tree/Tree';
+import { Tree, TreeFastPathResponse, TreeTableBody } from '@/types/tree/Tree';
 
 import { usePagination } from '@/hooks/usePagination';
 
-import { Skeleton } from '@/components/Skeleton';
-
 import TreeTable from '@/components/Table/TreeTable';
 
-import { useTreeTable } from '@/api/Tree';
+import { useTreeTable, useTreeTableFast } from '@/api/Tree';
 import { TableInfo } from '@/components/Table/TableInfo';
 
 import { formattedBreakLineValue } from '@/locales/messages';
 import { ItemsPerPageValues } from '@/utils/constants/general';
 
+import QuerySwitcher from '@/components/QuerySwitcher/QuerySwitcher';
+
 interface ITreeListingPage {
   inputFilter: string;
 }
 
+function isCompleteTree(
+  data: Tree | TreeFastPathResponse[number],
+): data is Tree {
+  return 'build_status' in data;
+}
+
 const TreeListingPage = ({ inputFilter }: ITreeListingPage): JSX.Element => {
-  const { origin: unsafeOrigin } = useSearch({ strict: false });
-  const origin = zOrigin.parse(unsafeOrigin);
   const [itemsPerPage, setItemsPerPage] = useState(ItemsPerPageValues[0]);
 
-  const { data, error, isLoading } = useTreeTable(origin);
+  const { data: fastData, status } = useTreeTableFast();
+
+  const { data, error, isLoading } = useTreeTable({
+    enabled: status === 'success' && !!fastData,
+  });
```

Comment:
Have you considered the possibility of doing it in parallel?

Reply:
I think it is better if is in series because we want the fast path ASAP, if the fast path is taking time to load, in parallel will stress an already stressed database